### DISCUSSION TITLE
Upgrade path from 14.0.0.0 to latest

### DISCUSSION THREAD

**Discussion by bobMuk** [awawa-dev/HyperHDR#366] [General]
Hi - I installed HyperHDR 14.0.0.0 on raspbian a few years ago and it works great.  I want to upgrade to the current latest (18.0.0.0) to get the new features.  A bit of googling suggests it's not a simple case of installing the latest .deb files, there's dependencies with Buster vs Bullseye.
Is there a (relatively) straightfoward way for me to go from 14 to 18 and retain my config?

**Comment by awawa-dev** [comment]
Hi
Yes, you will need Debian Bullseye system to make v18 work.
The safest path is to:
1) Backup your .hyperion folder that was used in v14 in your home directory (/home/pi/?). In next versions it's ```.hyperhdr``` folder.
2) Uninstall v14 and install v17 using manual https://github.com/awawa-dev/HyperHDR/wiki/Installation#Linux Adjust the download link and the file name from v18 to v17 version.
3) Run HyperHDR and check if all the settings were imported properly.
4) If everything is OK then export your configuration using ```General``` tab in HyperHDR v17 webpage. You will receive an JSON that contains all your settings in one file. Download it and save it.
5) Flash your SD card using SD image from the HyperHDR v18 release section. If you have Rpi 3 or 4 select aarch64 version.
6) Connect to your new installed v18 and in the general tab import your saved configuration. HyperHDR will restart automatically.
7) Consider to use new dedicated LUT tables from v18 that provide much better HDR tone mapping or calibrate new one using automatic wizard. It's described in v18 release info and on the blog.

**Reply by bobMuk** [reply]
Thanks for the detailed response!  I'd assumed that it wouldn't be as simple as exporting my settings from my v14 install, reflashing the SD with the v18 image and then importing them (I guess the export format from v14 isn't compatible with v18?)

I might just flash the v18 image onto a card and manually recreate my config since it's pretty standard besides my LED configuration.

And yes - the custom LUT tables are exactly why I want to upgrade.  Thank you for all the work that you've put into them!

**Reply by awawa-dev** [reply]
v18 won't find your current settings in your home folder because it's a very old v14 structure, but it should work for v17. And v17 allows finally to easy export all settings from web GUI. Upgrading to v17 should take you only few minutes and you don't need to re-flash the SD card to complete it.

**Reply by bobMuk** [reply]
I've got the option in the gui on v14 to export my settings into a json file.

Would it be possible for me to:
1) export configuration from the v14 gui
2) flash a new sd card with v18
3) replace the old card with the new one
4) boot up and import the json file that I exported?

I like this aproach since if something's not working I could just revert to the old SD card

**Reply by awawa-dev** [reply]
v14 was using not reliable export/import feature inherited from HyperionNG (e.g. you can export only one instance, I don't remember exactly but your grabbers settings may not export either) before it was re-implemented from scratch in HyperHDR in v17.
v14 file export format is not compatible with v17/v18.

**Reply by bobMuk** [reply]
Understood. Thanks for the clarification!

**Reply by chesterflaps** [reply]
> Hi Yes, you will need Debian Bullseye system to make v18 work. The safest path is to:
> 
> 1. Backup your .hyperion folder that was used in v14 in your home directory (/home/pi/?). In next versions it's `.hyperhdr` folder.
> 2. Uninstall v14 and install v17 using manual https://github.com/awawa-dev/HyperHDR/wiki/Installation#Linux Adjust the download link and the file name from v18 to v17 version.
> 3. Run HyperHDR and check if all the settings were imported properly.
> 4. If everything is OK then export your configuration using `General` tab in HyperHDR v17 webpage. You will receive an JSON that contains all your settings in one file. Download it and save it.
> 5. Flash your SD card using SD image from the HyperHDR v18 release section. If you have Rpi 3 or 4 select aarch64 version.
> 6. Connect to your new installed v18 and in the general tab import your saved configuration. HyperHDR will restart automatically.
> 7. Consider to use new dedicated LUT tables from v18 that provide much better HDR tone mapping or calibrate new one using automatic wizard. It's described in v18 release info and on the blog.

You make reference to V17 and then later V18 which is very confusing. I quote _**"install v17 using manual https://github.com/awawa-dev/HyperHDR/wiki/Installation#Linux Adjust the download link and the file name from v18 to v17 version."_**

You then say _"Flash your SD card using SD image from the HyperHDR v18 release section"_ and then "_Connect to your new installed v18_"

So do we get the v18 or v17?

Also, you said in point 1 to backup hyperion folder (will it not be hyperhdr??) but to do this do we need to remove the SD card from the Pi or do we access it a different way?

Sorry for all the questions but I built my Hyperhdr setup over a year ago and followed an instructional video so I'm new to updating the software but I know I need to. It'd be great to have a full step by step explanation assuming the person reading it knows nothing.

I'm upgrading from V16 btw.

Thanks

**Reply by awawa-dev** [reply]
> You make reference to V17 and then later V18 which is very confusing. I quote **"install v17 using manual https://github.com/awawa-dev/HyperHDR/wiki/Installation#Linux Adjust the download link and the file name from v18 to v17 version."**
You then say "Flash your SD card using SD image from the HyperHDR v18 release section" and then "Connect to your new installed v18"

Everything is correct: first you need to update v14-v16 to v17 (you cannot upgrade to v18 immediately because you need Bullseye OS). Versions before v17 had leaky web export system inherited from Hyperion (e.g. could export only single instance). Since v17 I rebuilt it completely and now it's up to common standards and also support multi-instances configurations.  So in point nr (4) you should have HyperHDR upgraded to v17 and that version allows you to export the settings from the www panel.  

>  Also, you said in point 1 to backup hyperion folder (will it not be hyperhdr??) but to do this do we need to remove the SD card from the Pi or do we access it a different way?  

That right's. In previous versions we completely broke with old naming. In v16 you have the ```.hyperhdr``` folder. Its exactly location you can find in your HyperHDR logs just after it starts. It's important to confirm it because you follow that tutorial and I'm not 100% sure how your installation ended up. You don't need to remove SD card and if you have ssh enabled on Rpi (default for HyperHDR images) you can connect to your Rpi using e.g. ```winscp.``` Little warning: folders begin with '.' like .hyperhdr are hidden and you should enable the option in ```winscp``` to show them. Then you can easily copy the folder to your PC to have a safe backup. Probably you won't need it if everything goes according to the plan and the backup exported from v17 panel in point (4) should be enough.

**Reply by chesterflaps** [reply]
I have Putty as an SSH client but I can;t seem to work out how to access files once connected to the Pi

**Reply by chesterflaps** [reply]
I've removed the SD card and put it into my pc and there are no directories or files which you mention. I'm completely lost

**Reply by chesterflaps** [reply]
When I plug the SD card in, I have 

boot (E:) I can access this and there's loads of files and a folder called overlays
USB Drive (F:) I cannot access this. The PC tells me it needs to be formatted as it is not a recognised format

**Reply by awawa-dev** [reply]
You must remember that Raspberry Pi OS root partition is Linux ext-type, so you need PC running a Linux to read it. If you didn't use official HyperHDR SD image (because SSH is always enabled) then you can follow typical Raspberry Pi OS manual how to enable it manually (a keyboard and a monitor will be needed for Rpi)

**Reply by chesterflaps** [reply]
I think I am going to start from scratch as I have made a note of my settings in V16. Where do I download the V18 software for Windows from?

**Reply by awawa-dev** [reply]
It is alongside other installers for Rpi / Linux / macOS in the release section.

**Reply by chesterflaps** [reply]
Ok I'm getting there. So when I put the wifi details on the SD card, do I leave or remove the" and the < that are there? I am trying all versions of having them and not having them and I cannot get it to work. I cannot get into it using SSH

**Reply by awawa-dev** [reply]
Leave " remove <
You connect to 5g wifi? Only Rpi4/3b+ can do it.

**Reply by awawa-dev** [reply]
You can apply wpa_supplicant.conf settings only once. After Rpi first boot further changes to this file will be ignored.

**Reply by chesterflaps** [reply]
i have 3b+ but my network is split off into 5g and 2.4

**Reply by chesterflaps** [reply]
it's been working fine on this network using v16

**Reply by chesterflaps** [reply]
> You can apply wpa_supplicant.conf settings only once. After Rpi first boot further changes to this file will be ignored.

I keep removing the sd card and redoing the wpaconfig part

**Reply by chesterflaps** [reply]
sorry, wpasupplicant

**Reply by awawa-dev** [reply]
To apply new wpasupplicant settings if the previous were wrong, you need to write the SD image again.

**Reply by chesterflaps** [reply]
so format the sd card and write the pi imager again?

**Reply by awawa-dev** [reply]
Yes, use Rufus.

**Reply by chesterflaps** [reply]
I used the Raspberry Pi Imager as I did when I first built my setup. Redone that, redone the wifi name and password and now waiting for the Pi to boot up and see if I can get into it

**Reply by chesterflaps** [reply]
Still not allowing me to login to it and not showing on my list of connected devices on my network

**Reply by awawa-dev** [reply]
Raspberry Pi Imager could mess up with the user credential, also has it's own setting for SSH. If something was working in the past, it doesnt mean it works now because Rapsberry Pi OS removed built-in user in April. Only Rufus is officially supported, you use Rpi Imager on your own risk.

**Reply by chesterflaps** [reply]
ok lets try rufus

**Reply by chesterflaps** [reply]
where on earth do i get rufus? There is a rufus.ie site which looks dodgy and when I click the v3.20 it opens up an ad and then goes to a page that won't load

**Reply by awawa-dev** [reply]
Here on github? https://github.com/pbatard/rufus

**Reply by chesterflaps** [reply]
Thank you. Following the instructions it says to download and extract the SD image from the hyperhdr github repositoryand then gives SD-card-image-18.0.0.0beta3-armv7l.img.xz but what is that?where do it get that?

**Reply by awawa-dev** [reply]
7-zip to extract?

**Reply by chesterflaps** [reply]
I have 7 zip but where do i get the file in the first place?

**Reply by awawa-dev** [reply]
It's all in the release section alongside all other installers

**Reply by chesterflaps** [reply]
I think I found it but not sure which to get as I have the 3b+

**Reply by chesterflaps** [reply]
It doesn't make reference to the 3B+

**Reply by awawa-dev** [reply]
https://github.com/awawa-dev/HyperHDR/wiki/Installation#raspberrypi-os-sd-card-image-installation

**Reply by chesterflaps** [reply]
thanks but it does not mention a 3B+. It mentions a Pi 3 in 2 parts so not sure which to go for

**Reply by awawa-dev** [reply]
same path as Rpi 3

**Reply by chesterflaps** [reply]
but the 3B+ is 64bit and the pi3 path says 32bit?

**Reply by awawa-dev** [reply]
you can use any of them

**Reply by chesterflaps** [reply]
Next issue, when I open rufus, it finds the SD card but where I should have the Hyperhdr under boot selection but it just says Disk or ISO image.... and there is no option to select the hyperhdr file. Once I have extracted it, does it need transferring to the SD card?

**Reply by chesterflaps** [reply]
Does the extracted file need to be put on the sd card before using rufus?

**Reply by awawa-dev** [reply]
Follow the manual in the wiki, you write the sd image and put additionally wpasupplicant only.

**Reply by chesterflaps** [reply]
I am following it and I cannot work it out. I downloaded the file, extracted it using 7-zip, transferred the extracted ISO file to the SD card and then opened Rufus only to see the same as before.  _"when I open rufus, it finds the SD card but where I should have the Hyperhdr under boot selection but it just says Disk or ISO image.... and there is no option to select the hyperhdr file."_

**Reply by awawa-dev** [reply]
You need Rufus to flash the img file to SD card as on the example on screenshot in the wiki, not copy it to sd

**Reply by chesterflaps** [reply]
Ok so I think I got it to work and the image has written to the SD card but now I cannot access the SD card

**Reply by chesterflaps** [reply]
Windows tells me I need to format it to access it so how do I add the wpa_supplicant into it?

**Reply by chesterflaps** [reply]
I do not have a 'boot folder'

**Reply by chesterflaps** [reply]
The instructions say nothing about a boot folder

**Reply by awawa-dev** [reply]
You flashed img file extracted from xy archive? eject/insert againt sd card. I have no other ideas

**Reply by chesterflaps** [reply]
Not sure what you mean by flashed img file. I extracted it using 7-zip as the instructions say and then used rufus to put it onto the sd card

**Reply by awawa-dev** [reply]
you extract xy to have img and that file you use with Rufus

**Reply by chesterflaps** [reply]
extract xy? what is that? I downloaded this file from github [SD-card-image-18.0.0.0-armv7l.img.xz] as per the instructions for the Raspberry Pi 3. I then used 7 zip to extract it and then used rufus to put it onto the SD card

**Reply by chesterflaps** [reply]
Does writing it to the sd card using rufus, create a boot folder or do I have to create that folder before I write to it? Or do I partition the SD card first? There is no mention of this in the instructions

**Reply by awawa-dev** [reply]
yes I misspelled because I'm getting tired ;) 
![obraz](https://user-images.githubusercontent.com/69086569/200958703-ad1a7342-89b7-443a-a68b-3b745d7708a3.png)
If you put img file in the boot selection and flash it to sd then everything should be OK. At least it works for me and other users. After finish you should have boot partition and another one (Windows claims it's un-formatted but it's Linux partition)

**Reply by chesterflaps** [reply]
Ok so if windows says its unformatted, how do I access it on a windows pc?

**Reply by awawa-dev** [reply]
there are **two** partitions

**Reply by chesterflaps** [reply]
Yeah it's not working and won't let me access the sd card. I've followed the instructions twice and it does not work

**Reply by chesterflaps** [reply]
I'm going to try the other img download to see if that works

**Reply by chesterflaps** [reply]
Nope, same again, not accessible

**Reply by chesterflaps** [reply]
followed the instructions 3 times and failed every time

**Reply by chesterflaps** [reply]
You said "_If you put img file in the boot selection and flash it to sd then everything should be OK"_

What is the boot selection. Please do not assume I know what you are talking about because clearly I have no idea

**Reply by awawa-dev** [reply]
You dont need test all images. Everything is fine with them, including SD-card-image-18.0.0.0-armv7l.img
![obraz](https://user-images.githubusercontent.com/69086569/200963498-b3e8671b-78e8-4f60-b35e-db14101988b9.png)
Windows Computer management tab.
Two partitions as I said. One is boot.

**Reply by chesterflaps** [reply]
Mine has the same but i cannot get into the boot partition

**Reply by chesterflaps** [reply]
![image](https://user-images.githubusercontent.com/86977015/200964139-99aa7309-3606-4bb6-a000-2d8e63307230.png)

**Reply by awawa-dev** [reply]
so maybe assign a letter for a drive if Windows didn't do it for you?

**Reply by chesterflaps** [reply]
It is labelled as E on windows

**Reply by awawa-dev** [reply]
If doesnt have a letter on your screenshot

**Reply by chesterflaps** [reply]
ok so getting closer. I have the SD card in the PI and can see it is connected to my wifi. Now I cannot see the same ELD selection as I had before. Before I had ws281x but that is not there to select

**Reply by chesterflaps** [reply]
In fact I cannot select anything. It thinks I have a HD camera for image capture but I sue a capture card

**Reply by chesterflaps** [reply]
Got it working (kind of). Turns out I needed to go to a different webpage for the HyperHDR instead of settings from the hyper hdr windows app. I say kind of because it keeps randomly going off.

Thank you for your patience with me today

**Comment by n0valis** [comment]
Very useful info.
Should be more visible. Maybe pin or link somewhere at the frontpage.
